### ISSUE TITLE
Crash on Java wrapper when using gen3d

### ISSUE THREAD

**Issue by niohiki** [openbabel/openbabel#1854]
I am using OpenBabel 2.4.1. I am using the Java wrapper to transform SMILES to MOL. The following code crashes the JVM, without an Error.

`public class OBTest {

    public static void main(String[] args) {
        System.loadLibrary("openbabel_java");

        OBConversion conversion = new OBConversion();
        OBOp gen = OBOp.FindType("gen3d");
        OBMol mol = new OBMol();
        
        conversion.SetInFormat("smi");
        conversion.ReadString(mol, "CC");
        
        gen.Do(mol);

        conversion.SetOutFormat("mol");
        System.out.println(conversion.WriteString(mol));
    }
}`

The problematic line is

`gen.Do(mol);`

If it is removed, or if `gen2d` is used in

`OBOp gen = OBOp.FindType("gen2d");`

the code works correctly. I have repeated the same steps on the GUI and the command line, and it works perfectly.

**Comment by baoilleach** [comment]
You need to check the return value of the earlier API calls. My guess is
that some are false.

**Comment by niohiki** [comment]
Both SetInFormat and ReadString return true. This must be the case, because as noted, when using "gen2d" the code works perfectly fine, and outputs the mol data with the 2d positions.

**Comment by baoilleach** [comment]
Good point. Then my question is, is gen a NULL pointer? I'm not sure how this translates into Java.

**Comment by niohiki** [comment]
Nope, in fact I can even do

`System.loadLibrary("openbabel_java");

OBConversion conversion = new OBConversion();
OBOp gen3d = OBOp.FindType("gen3d");
OBMol mol = new OBMol();

System.out.println(gen3d.WorksWith(mol));
        
conversion.SetInFormat("smi");
conversion.ReadString(mol, "CC");

gen3d.Do(mol);

conversion.SetOutFormat("mol");
System.out.println(conversion.WriteString(mol));`

(notice the extra call to println) and the output is

`true`

and intermediately in the error stream

`Java Result: -1073740791`

so it's crashing the JVM pretty good. Something's going pretty wrong with the JNI interface because I would expect some JVM error, if not an Exception, to be thrown.

**Comment by baoilleach** [comment]
Fair enough. Can you provide details of your compiler, SWIG, OS, Java version, and anything else you think we might need to reproduce it.

**Comment by niohiki** [comment]
Sure! 

JDK: 1.8.0_73 (JRE from JDK)
OS: Windows 10

I do not directly need SWIG - I am using the OpenBabel binaries for windows, version 2.4.1. I load both the native library and the jar from that same OpenBabel installation, of course. I am developing inside Netbeans, but just in case I compiled/run the code directly with the java tools directly from the prompt, and it fails anyway.

**Comment by baoilleach** [comment]
And the 64-bitness of everything?

**Comment by niohiki** [comment]
Both OpenBabel and JDK are 64 bits. And, the 32 bit dll cannot be loaded into 64 bit java.

**Comment by serval2412** [comment]
I tried to give it a try but just fail to even launch the class.
I got this:
Exception in thread "main" java.lang.UnsatisfiedLinkError: no openbabel_java in java.library.path
whereas I followed http://openbabel.org/docs/dev/UseTheLibrary/Java.htmlt:
LD_LIBRARY_PATH=/usr/local/lib/libopenbabel_java.so
CLASSPATH=/home/julien/projects/openbabel/build/scripts/java/openbabel.jar:.

(I'm on Linux Debian x86-64)
